Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -887,26 +887,24 @@ RefPtr<Statement> Parser::try_parse_labelled_statement(AllowLabelledFunction all
         if (function_declaration->kind() == FunctionKind::Async)
             syntax_error("Async functions cannot be defined in labelled statements");
 
-        labelled_statement = move(function_declaration);
+        labelled_item = move(function_declaration);
     } else {
         m_state.labels_in_scope.set(identifier, {});
-        labelled_statement = parse_statement(allow_function);
-        if (is<IterationStatement>(*labelled_statement)) {
+        labelled_item = parse_statement(allow_function);
+        if (is<IterationStatement>(*labelled_item))
             is_iteration_statement = true;
-            static_cast<IterationStatement&>(*labelled_statement).add_label(identifier);
-        } else if (is<LabelableStatement>(*labelled_statement)) {
-            static_cast<LabelableStatement&>(*labelled_statement).add_label(identifier);
-        }
     }
 
-    if (!is_iteration_statement) {
+    // Only check if this is the innermost (i.e. the labelled item is not another LabelledStatement),
+    // possibly nested labelled statement, which has access to all the preceding labels via labels_in_scope.
+    if (!is_iteration_statement && !is<LabelledStatement>(*labelled_item)) {
         if (auto entry = m_state.labels_in_scope.find(identifier); entry != m_state.labels_in_scope.end() && entry->value.has_value())
             syntax_error("labelled continue statement cannot use non iterating statement", m_state.labels_in_scope.get(identifier).value());
     }
```

Comment:
Although this passes all test262 and our tests this is not quite correct. We have to check that the labelled statement.

Now a non iteration statement nested within an iteration statement (or the other way around) passes while it should fail. With some having multiple labels.
Examples:
```js
outer: outer2: {
    for (; false;)
        continue outer;
}

for (;false;) outer: outer2: {
    continue outer;
}
```

These should probably also become tests.

Reply:
Should be fixed, thanks!